parse/parser.go

```diff
@@ -381,6 +381,77 @@ func parseStringLiteral(buf TokenBuffer) (ast.Expression, error) {
 	return &ast.StringLiteral{Value: token.Val}, nil
 }
 
+func parseFunctionLiteral(buf TokenBuffer) (*ast.FunctionLiteral, error) {
+	lit := &ast.FunctionLiteral{}
+	var err error
+
+	if err = expectNext(buf, Function); err != nil {
+		return nil, err
+	}
+
+	tok := buf.Read()
+	if tok.Type != Ident {
+		return nil, parseError{
+			tok.Type,
+			"token is not identifier",
+		}
+	}
+	lit.Name = &ast.Identifier{Value: tok.Val}
+
+	if err = expectNext(buf, Lparen); err != nil {
+		return nil, err
+	}
+
+	if lit.Parameters, err = parseFunctionParameters(buf); err != nil {
```

Comment:
I think it has some bug when function has no parameter, ex) func a () {}
After checking Lparen, parseFunctionParameters does not read Lparen from buf and checks Rparen, so it can not filter out empty parameters.

I ran the following test case (1) in your branch function TestParseFunctionLiteral and found the error (2).

(1)
``` go
{
	// fun example () {}
	{Type: Function, Val: "func"},
	{Type: Ident, Val: "example"},
	{Type: Lparen, Val: "("},
	{Type: Rparen, Val: ")"},
	{Type: Lbrace, Val: "{"},
	{Type: Rbrace, Val: "}"},
}
```

(2)
```
--- FAIL: TestParseFunctionLiteral (0.00s)
    parser_internal_test.go:522: test[1] - TestParseFunctionLiteral wrong result
        expected: func name(Parameter : (Identifier: a, Type: int), Parameter : (Identifier: b, Type: string))  {
        int [IDENT, c] = 5
        }
        got: func example()  {
        
        }
FAIL
```

Reply:
Therefore, I hope to add a TC for the empty parameter. :)

Reply:
Ok I will check it :-)

Reply:
no i think my mistake

Reply:
expectNext has buf.read().I could not catch it.

Reply:
And plz add comment of this function

Reply:
ok